### PR TITLE
fix service to work with DTO instead of Entity

### PR DESCRIPTION
**Pull request by AndriiMovhan**
fix service to work with DTO instead of Entity

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -75,6 +75,13 @@
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/org.hibernate/hibernate-core -->
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-core</artifactId>
```
Comment (financeasist) [inline pom.xml:81]: why do we need this?

File: src/main/java/com/example/shop/service/GoodServiceImpl.java [hunk 2]
```diff
@@ -30,13 +31,14 @@
...
-        return goodRepository.save(good);
+    @Transactional
+    public GoodDto save(GoodDto goodDto) {
+        return goodMapper.toDto(goodRepository.save(goodMapper.toEntity(goodDto)));
     }
 
     @Override
-    public Good update(Good id) {
-        return goodRepository.save(id);
+    public GoodDto update(GoodDto id) {
```
Comment (financeasist) [inline src/main/java/com/example/shop/service/GoodServiceImpl.java:40]: please add `@Transactional` to all methods that modifies an entity and `@Transactional(readOnly = true)`  to all find methods

Comment (financeasist) [inline src/main/java/com/example/shop/service/GoodServiceImpl.java:40]: `update(GoodDto id)`  ?? is it really valid name here?

**Review by financeasist — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (AndriiMovhan) [reply pom.xml:81]: I fix this

File: src/main/java/com/example/shop/service/GoodServiceImpl.java (hunk 2 above)
Comment (AndriiMovhan) [reply src/main/java/com/example/shop/service/GoodServiceImpl.java:40]: I add this

Comment (AndriiMovhan) [reply src/main/java/com/example/shop/service/GoodServiceImpl.java:40]: I fix this

Comment (AndriiMovhan) [reply src/main/java/com/example/shop/service/GoodServiceImpl.java:42]: I fix this

**Pull request merged**